adapters/AppLovin/AppLovinAdapter/GADMediationAdapterAppLovin.m

```diff
@@ -20,53 +20,65 @@
+  GADMAdapterAppLovinRewardedRenderer *_rewardedRenderer;
+}
 
-+ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
-             completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // Compile all the SDK keys we should initialize for
++ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
+             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
+  // Compile all the SDK keys that should be initialized.
   NSMutableSet<NSString *> *sdkKeys = [NSMutableSet set];
   NSObject *sdkKeysLock = [[NSObject alloc] init];
 
-  // Compile SDK keys from configuration credentials
+  // Compile SDK keys from configuration credentials.
   for (GADMediationCredentials *credentials in configuration.credentials) {
-    NSString *sdkKey = credentials.settings[GADMAdapterAppLovinConstant.sdkKey];
-    if (sdkKey) {
-      [sdkKeys addObject:sdkKey];
+    NSString *sdkKey = credentials.settings[GADMAdapterAppLovinSDKKey];
+    if ([GADMAdapterAppLovinUtils isValidAppLovinSDKKey:sdkKey]) {
+      GADMAdapterAppLovinMutableSetAddObject(sdkKeys, sdkKey);
     }
   }
 
-  // Add SDK key from Info.plist if exists
-  if ([GADMAdapterAppLovinUtils infoDictionaryHasValidSDKKey]) {
+  // Add SDK key from Info.plist if it exists.
+  if ([GADMAdapterAppLovinUtils infoDictionarySDKKey]) {
     NSString *sdkKey = [GADMAdapterAppLovinUtils infoDictionarySDKKey];
-    [sdkKeys addObject:sdkKey];
+    if ([GADMAdapterAppLovinUtils isValidAppLovinSDKKey:sdkKey]) {
+      GADMAdapterAppLovinMutableSetAddObject(sdkKeys, sdkKey);
+    }
   }
 
-  // Log to the publisher that having more than 1 SDK key is... abnormal
-  if (sdkKeys.count > 1) {
-    [GADMAdapterAppLovinUtils
-        log:@"Found %lu SDK keys. Please remove any SDK keys you are not using from the AdMob UI.",
-            sdkKeys.count];
+  // Log to the publisher that having more than 1 SDK key is... abnormal.
+  if (!sdkKeys.count) {
+    NSString *errorString = @"No SDK keys are found. Please add valid SDK keys in the AdMob UI.";
+    [GADMAdapterAppLovinUtils log:errorString];
+    NSError *error =
+        GADMAdapterAppLovinErrorWithCodeAndDescription(kGADErrorMediationAdapterError, errorString);
+    completionHandler(error);
+    return;
   }
 
-  // Initialize SDKs based on SDK keys
+  [GADMAdapterAppLovinUtils
+      log:@"Found %lu SDK keys. Please remove any SDK keys you are not using from the AdMob UI.",
+          (unsigned long)sdkKeys.count];
+
+  // Initialize SDKs based on SDK keys.
   NSSet<NSString *> *sdkKeysCopy = [sdkKeys copy];
   for (NSString *sdkKey in sdkKeysCopy) {
     [GADMAdapterAppLovinUtils log:@"Initializing SDK for SDK key %@", sdkKey];
 
     ALSdk *sdk = [GADMAdapterAppLovinUtils retrieveSDKFromSDKKey:sdkKey];
     [sdk initializeSdkWithCompletionHandler:^(ALSdkConfiguration *configuration) {
       @synchronized(sdkKeysLock) {
```

Comment:
We don't need this `@synchronized` block anymore if we're using the new atomic functions now.

Reply:
removed